overlord/patch/patch_test.go

```diff
@@ -456,3 +457,31 @@ func generatePatchFunc(testValue int, sequence *[]int) patch.PatchFunc {
 		return nil
 	}
 }
+
+func (s *patchSuite) TestRegressionCoreCurrentSymlinkMissing(c *C) {
+	// simulate missing "core" symlink
+	dirs.SetRootDir(c.MkDir())
+	defer func() { dirs.SetRootDir("/") }()
+
+	log, restore := logger.MockLogger()
+	defer restore()
+
+	st := state.New(nil)
+	patch.Init(st)
+
+	st.Lock()
+	siCore1 := &snap.SideInfo{RealName: "core", Revision: snap.R(1)}
+	siCore2 := &snap.SideInfo{RealName: "core", Revision: snap.R(2)}
+	snapstate.Set(st, "core", &snapstate.SnapState{
+		SnapType: "os",
+		Active:   true,
+		Sequence: []*snap.SideInfo{siCore1, siCore2},
+		Current:  siCore1.Revision,
+	})
+	st.Unlock()
+
+	subLevel := 1
+	err := patch.MaybeResetSublevelForLevel60(st, &subLevel)
+	c.Assert(err, IsNil)
+	c.Assert(log.String(), Matches, `(?m).*WARNING: cannot determine core refresh time: lstat /.*/snap/core/current: no such file or directory`)
```

Comment:
Could you please also add asserts to check that subLevel argument was set back to 0, and "patch-sublevel" and "patch-sublevel-reset" in the state were updated (per the comment I suggested above)?